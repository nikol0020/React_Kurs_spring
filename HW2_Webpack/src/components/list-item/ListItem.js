import React, {Component} from 'react';
import './ListItem.scss';

import PropTypes from 'prop-types';

const ListItem = (props) => {

    const {id, title, genres, release_date, poster_path} = props;

    return (

        <div className="list-item" key={ id }>
            <img src={poster_path} alt="picture" className="list-item__image"/>
            <div className="list-item__short-description">
                <h4 className="list-item__title">{ title }</h4>
                <span className="list-item__genre">{ genres }</span>
                <span className="list-item__year">{ release_date }</span>
            </div>
        </div>
    )
};

ListItem.propTypes = {
    id: PropTypes.number,
    title: PropTypes.string,
    genres: PropTypes.string,
    release_date: PropTypes.string,
    poster_path: PropTypes.string
};

export default ListItem;
