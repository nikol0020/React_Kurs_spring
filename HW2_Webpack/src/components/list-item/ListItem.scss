.list-item {
  width: 200px;
  height: 430px;
  background-color: #fff;
  margin: 20px;
  color: #333;
  border: 2px solid #a3a;

  &__title {
    margin: 10px 0;
    text-align: center;
  }

  &__image {
    width: 100%;
    height: 315px;
    min-width: 200px;
    min-height: 300px;
    background-color: #bbf;
  }

  &__short-description {
    margin: 10px;
  }

  &__genre {
    float: left;
    overflow: hidden;
    text-overflow: ellipsis;
    height: 4.5vh;
  }

  &__year {
    float: right;
    color: #a3a;
  }

}
