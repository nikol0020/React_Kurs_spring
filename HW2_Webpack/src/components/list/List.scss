.list {
  display: flex;

  &__item{
    list-style: none;
  }

}
