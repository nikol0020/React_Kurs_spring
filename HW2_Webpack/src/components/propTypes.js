import PropTypes from 'prop-types';

const movieShape = PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    tagline: PropTypes.string,
    vote_average: PropTypes.number,
    vote_count: PropTypes.number,
    release_date: PropTypes.string,
    poster_path: PropTypes.string,
    overview: PropTypes.string,
    budget: PropTypes.number,
    revenue: PropTypes.number,
    genres: PropTypes.arrayOf(PropTypes.string),
    runtime: PropTypes.number
});

const propTypes = {
    data: PropTypes.arrayOf(movieShape)
};

export default propTypes;
