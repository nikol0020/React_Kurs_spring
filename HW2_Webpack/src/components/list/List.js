import React from "react";
import './List.scss';
import '../propTypes';
import ListItem from '../list-item';

const List = (props) => {

    const elements = props.data.map((item) => {

        const {id, ...itemProps} = item;
        return (
            <li key={id}>
                <ListItem {...itemProps} > </ListItem>
            </li>
        )
    });

    return (
        <ul className="list">
            {elements}
        </ul>
    )
};

export default List;