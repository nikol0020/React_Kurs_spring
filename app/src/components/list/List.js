import React from "react";
import './List.scss';
import '../propTypes';
import ListItem from '../list-item';
import ErrorPage from '../error-page';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {select} from '../../store/actions/index';
import {Link, Redirect} from 'react-router-dom';

const List = (props) => {

    if (props.error) {
        return (
            <Redirect to = {'/404'}/>
        )
    }

    // initialState = {
    //     movies: [],
    //     loading: false,
    //     error: null
    // };

    const elements = () => {
        return props.movies.map((item) => {

            const {id, ...itemProps} = item;
            return (
                <li key={id}>
                    <Link to ={`/movie/${id}`}>
                        <ListItem {...itemProps} > </ListItem>
                    </Link>
                </li>
            )
        });
    };

    return (
        <ul className="list">
            {elements()}
        </ul>
    )
};

function mapStateToProps(state) {
    return {
        movies: state.searchFilter.movies ? state.searchFilter.movies : state.movies.movies,
        error: state.movies.error

    };
};

function matchDispatchToProps(dispatch) {
    return bindActionCreators({select: select}, dispatch)
};

export default connect(mapStateToProps, matchDispatchToProps)(List);
