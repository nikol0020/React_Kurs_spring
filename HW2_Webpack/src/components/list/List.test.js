import React from 'react';
import List from './List';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import '../propTypes';
import ListItem from '../list-item';

configure({ adapter: new Adapter() });

describe('<List /> shallow rendering', () => {

    it('should render a ul', () => {
        const wrapper = shallow(
            <ul className="list"></ul>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a id', () => {
        const wrapper = shallow(
            <li id="id"></li>
        );
        expect(wrapper.prop('id')).toEqual('id');
    });

});