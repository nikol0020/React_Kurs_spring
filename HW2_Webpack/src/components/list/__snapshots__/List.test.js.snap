// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> shallow rendering should render a ul 1`] = `
<ul
  className="list"
/>
`;
